CorsixTH/Src/th_map.cpp

```diff
@@ -1600,12 +1601,12 @@ void level_map::depersist(lua_persist_reader* pReader) {
     if (!pReader->read_stack_object()) {
       return;
     }
-    pNode->next = (link_list*)lua_touserdata(L, -1);
-    if (pNode->next) {
-      if (pNode->next->prev != nullptr) {
+    pNode->entities.next = (link_list*)lua_touserdata(L, -1);
```

Comment:
Good, can you replace all these C style casts at the same time. Pretty sure all the pointer ones can be static_cast<>